Mage.Sets/src/mage/cards/t/T45PowerArmor.java

```diff
@@ -0,0 +1,107 @@
+    public T45PowerArmor(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}");
+        
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // When T-45 Power Armor enters the battlefield, you get {E}{E}.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new GetEnergyCountersControllerEffect(2), false));
+        // Equipped creature gets +3/+3 and doesn't untap during its controller's untap step.
+        Ability ability = (new SimpleStaticAbility(new BoostEquippedEffect(3,3)));
+        ability.addEffect(new DontUntapInControllersUntapStepEnchantedEffect().setText("and doesn't untap during its controller's untap step"));
+        this.addAbility(ability);
+        // At the beginning of your upkeep, you may pay {E}. If you do, untap equipped creature, then put your choice of a menace, trample or lifelink counter on it.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(
+                new DoIfCostPaid(
+                        new T45PowerArmorEffect(),
+                        new PayEnergyCost(1)
+                ),
+                TargetController.YOU,
+                false
+        ));
+
+        // Equip {3}
+        this.addAbility(new EquipAbility(3));
+    }
+
+    private T45PowerArmor(final T45PowerArmor card) {
+        super(card);
+    }
+
+    @Override
+    public T45PowerArmor copy() {
+        return new T45PowerArmor(this);
+    }
+}
+
+class T45PowerArmorEffect extends OneShotEffect {
+
+    private static final Set<String> choices = new LinkedHashSet<>(Arrays.asList("Menace", "Trample", "Lifelink"));
+
+    T45PowerArmorEffect() {
+        super(Outcome.BoostCreature);
+        staticText = "untap eqipped creature, then put your choice of a menace, trample, or lifelink counter on it";
+    }
+
+    protected T45PowerArmorEffect(mage.cards.t.T45PowerArmorEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        Permanent creature = game.getPermanent(source.getSourcePermanentIfItStillExists(game).getAttachedTo());
+        game.getPermanent(source.getSourcePermanentIfItStillExists(game).getAttachedTo()).untap(game);
+        if (player == null || creature == null) {
+            return false;
+        }
+
+        Choice choice = new ChoiceImpl(true);
+        choice.setMessage("Choose menace, trample, or lifelink");
+        choice.setChoices(choices);
+        player.choose(outcome, choice, game);
+        String chosen = choice.getChoice();
+        if (chosen != null) {
+            creature.addCounters(CounterType.findByName(
+                    chosen.toLowerCase(Locale.ENGLISH)
+            ).createInstance(), source.getControllerId(), source, game);
+        }
+
+        return true;
+    }
+
+    @Override
+    public mage.cards.t.T45PowerArmorEffect copy() {
```

Comment:
no need to specify mage.cards.t here. I recommend using the gen-card.pl script to take care of the boilerplate

Reply:
I did use gen-card.pl not sure if it did something weird or if it got changed inadvertently while editing.

Reply:
ah good. yeah, this is a separate class not in scope of gen-card.pl, apologies for misleading. probably happened when it was copy-pasted from some other code as a template, just gotta double check everything you copy paste.